daemon/algod/api/server/v2/test/handlers_test.go

```diff
@@ -1263,6 +1264,11 @@ func TestTealDisassemble(t *testing.T) {
 	// Test bad program.
 	badProgram := []byte{1, 99}
 	tealDisassembleTest(t, badProgram, 400, "invalid opcode", true)
+
+	// Create a program with MaxTealSourceBytes+1 bytes
+	// This should fail inside the handler when reading the bytes from the request body.
+	largeProgram := []byte(strings.Repeat("a", v2.MaxTealSourceBytes+1))
+	tealDisassembleTest(t, largeProgram, 400, "http: request body too large", true)
```

Comment:
These tests are affected by a smaller size limit (200KB instead of 10MB) defined by `MaxTealSourceBytes` and will trigger the error here: https://github.com/algorand/go-algorand/blob/eb91eb5600472cd8f5a348f2481eda31b2df1025/daemon/algod/api/server/v2/handlers.go#L1003